src/leo_gateway_http_commons.erl

```diff
@@ -738,30 +738,76 @@ get_range_object(Req, Bucket, Key, {error, badarg}) ->
     ?reply_bad_range([?SERVER_HEADER], Key, <<>>, Req);
 get_range_object(Req, Bucket, Key, {_Unit, Range}) when is_list(Range) ->
     Mime = leo_mime:guess_mime(Key),
-    Header = [?SERVER_HEADER,
-              {?HTTP_HEAD_RESP_CONTENT_TYPE,  Mime}],
-    {ok, Req2} = cowboy_req:chunked_reply(?HTTP_ST_PARTIAL_CONTENT, Header, Req),
-    get_range_object_1(Req2, Bucket, Key, Range, undefined).
+    case get_body_length_1(Key, Range) of
+        {ok, Length} ->
+            Header = [?SERVER_HEADER,
+                      {?HTTP_HEAD_RESP_CONTENT_TYPE,  Mime},
+                      {?HTTP_HEAD_RESP_CONTENT_LENGTH,integer_to_list(Length)}],
+            Req2 = cowboy_req:set_resp_body_fun(
+                     fun(Socket, Transport) ->
+                             get_range_object_1(Req, Bucket, Key, Range, undefined, Socket, Transport)
+                     end, 
+                     Req),
+            ?reply_partial_content(Header, Req2);
+        {error, bad_range} ->
+            ?reply_bad_range([?SERVER_HEADER], Key, <<>>, Req);
+        {error, unavailable} ->
+            ?reply_service_unavailable_error([?SERVER_HEADER], Key, <<>>, Req);
+        _ ->
+            ?reply_not_found([?SERVER_HEADER], Key, <<>>, Req)
```

Comment:
Need to handle errors of the below clauses
- {error, ?ERR_TYPE_INTERNAL_ERROR}
- {error, timeout}